client/lib/domains/dns/index.js

```diff
@@ -126,7 +130,57 @@ function getFieldWithDot( field ) {
 	return ( typeof field === 'string' && field.match( /^([a-z0-9-_]+\.)+\.?[a-z]+$/i ) ) ? field + '.' : field;
 }
 
-module.exports = {
-	validateAllFields,
-	getNormalizedData
+function isWpcomRecord( record ) {
+	return startsWith( 'wpcom:', record.id );
+}
+
+function filterRootARecords( domain, records ) {
+	return filter( records, {
+		type: 'A',
+		name: `${domain}.`
+	} );
+}
+
+function removeDuplicateWpcomRecords( domain, records ) {
+	const wpcomRecord = find( filterRootARecords( domain, records ), isWpcomRecord ),
+		customRecord = find( records, { type: 'A', name: '' } );
+
+	let result;
```

Comment:
This could be simplified to just
```
 	if ( wpcomRecord && customRecord ) {
 		return without( records, wpcomRecord );
 	}

 	return records;
```
And no need for the `result` variable :) I know, this just me nitpicking, but in this case I believe the shorter version is more readable (you don't have to track what's the value of `result`).